File: ingestion/scripts/test_source_registry.py
from config.sources.registry import (
    get_active_sources,
    get_source_by_slug
)


def main():

    sources = get_active_sources()


    print(
        f"Active sources: {len(sources)}"
    )


    for source in sources:

        print(
            source["slug"],
            "|",
            source["name"],
            "|",
            source["type"]
        )


    hackernews = get_source_by_slug(
        "hacker-news"
    )


    print("\nHacker News source:")

    print({
        "slug": hackernews["slug"],
        "name": hackernews["name"],
        "type": hackernews["type"],
        "enabled": hackernews["enabled"]
    })


if __name__ == "__main__":
    main()

File: backend/scripts/testSourceRegistry.js
const {
    getSourceBySlug
  } = require("../config/sources");
  
  const run = async () => {
    const source =
      getSourceBySlug("hacker-news");
  
    if (!source) {
      throw new Error(
        "Hacker News source not found"
      );
    }
  
    console.log(
      "Source configuration:"
    );
  
    console.log({
      slug: source.slug,
      name: source.name,
      websiteUrl: source.websiteUrl,
      type: source.type
    });
  
    const stories =
      await source.scraper();
  
    console.log(
      `Stories fetched through source registry: ${stories.length}`
    );
  };
  
  run().catch((error) => {
    console.error(
      "Source registry test failed:",
      error.message
    );
  
    process.exit(1);
  });